### PR TITLE
[Enhancement] broker support multiple name services

### LINKED ISSUE #23509: [Enhancement] The broker supports multiple sets of name services configurations when creating external resources
## Enhancement
+ BEFORE
```
CREATE EXTERNAL RESOURCE "external_rs"
PROPERTIES (
"type" = "spark",
"spark.master" = "yarn",
"spark.submit.deployMode" = "cluster",
"spark.eventLog.enabled" = "true",
"spark.eventLog.compress" = "false",
"spark.eventLog.dir" = "hdfs://HCluster/tmp/spark/staging/historylog ",
"spark.history.fs.logDirectory" = "hdfs://HCluster/tmp/spark/staging/historylog",
"spark.yarn.queue" = "yarn_queue",
"spark.hadoop.yarn.resourcemanager.address.rm1" = "127.0.0.1:18040",
"spark.hadoop.yarn.resourcemanager.address.rm2" = "127.0.0.1:18040",
"spark.hadoop.yarn.resourcemanager.ha.enabled" = "true",
"spark.hadoop.yarn.resourcemanager.ha.rm-ids" = "rm1,rm2",
"spark.hadoop.yarn.resourcemanager.hostname.rm1" = "127.0.0.1",
"spark.hadoop.yarn.resourcemanager.hostname.rm2" = "127.0.0.1",
"spark.hadoop.dfs.nameservices" = "ACluster,HCluster",
"spark.hadoop.dfs.ha.namenodes.ACluster" = "r1,r2",
"spark.hadoop.dfs.client.failover.proxy.provider.ACluster" = "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider",
"spark.hadoop.dfs.namenode.rpc-address.ACluster.r1" = "127.0.0.1:8888",
"spark.hadoop.dfs.namenode.rpc-address.ACluster.r2" = "127.0.0.1:8888",
"spark.hadoop.dfs.ha.namenodes.HCluster" = "nn1,nn2",
"spark.hadoop.dfs.client.failover.proxy.provider.HCluster" = "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider",
"spark.hadoop.dfs.namenode.rpc-address.HCluster.nn1" = "127.0.0.1:8020",
"spark.hadoop.dfs.namenode.rpc-address.HCluster.nn2" = "127.0.0.1:8020",
"broker.dfs.client.federation.router.ns" = "ACluster",
"broker.dfs.ha.namenodes.ACluster" = "r1,r2",
"broker.dfs.client.failover.proxy.provider.ACluster" = "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider",
"broker.dfs.namenode.rpc-address.ACluster.r1" = "127.0.0.1:8888",
"broker.dfs.namenode.rpc-address.ACluster.r2" = "127.0.0.1:8888",
"broker.dfs.ha.namenodes.HCluster" = "nn1,nn2",
"broker.dfs.client.failover.proxy.provider.HCluster" = "org.apache.hadoop.hdfs.server.namenode.ha.ConfiguredFailoverProxyProvider",
"broker.dfs.namenode.rpc-address.HCluster.nn1" = "127.0.0.1:8020",
"broker.dfs.namenode.rpc-address.HCluster.nn2" = "127.0.0.1:8020",
"spark.hadoop.fs.defaultFS" = "hdfs://HCluster",
"working_dir" = "hdfs://HCluster/user/xx/sparketl",
"broker" = "hdfs_broker1",
"broker.username" = "xxx",
"broker.password" = "yyy"
)
```
> throw exception
```
cause by: load request missed necessary arguments for ha mode
```
+ AFTER
pass

### PR DESCRIPTION
**Pull request by blanklin030**
## Problem Summary:
Fixes #23509

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

File: fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java [hunk 1]
```diff
@@ -390,48 +390,53 @@ public BrokerFileSystem getDistributedFileSystem(String scheme, String path, Map
...
+                    for (String dfsNameService: dfsNameServiceArray) {
+                        // ha hdfs arguments
+                        dfsNameService = dfsNameService.trim();
+                        final String dfsHaNameNodesKey = DFS_HA_NAMENODES_PREFIX + dfsNameService;
+                        if (!properties.containsKey(dfsHaNameNodesKey)) {
+                            throw new BrokerException(TBrokerOperationStatusCode.INVALID_ARGUMENT,
+                                    "load request missed necessary arguments for ha mode");
+                        }
+                        String dfsHaNameNodes = properties.get(dfsHaNameNodesKey);
+                        conf.set(DFS_NAMESERVICES_KEY, dfsNameService);
```
Comment (wyb) [inline fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java:404]:
```suggestion
                        conf.set(DFS_NAMESERVICES_KEY, dfsNameServices);
```

Comment (wyb) [reply fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java:404]: should set `DFS_NAMESERVICES_KEY` before for loop?

**Review by Astralidea — DISMISSED** [review]

File: fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java [hunk 2]
```diff
@@ -390,48 +390,52 @@ public BrokerFileSystem getDistributedFileSystem(String scheme, String path, Map
...
-                            String nameNodeRpcAddress =
-                                    DFS_HA_NAMENODE_RPC_ADDRESS_PREFIX + dfsNameServices + "." + nameNode;
-                            if (!properties.containsKey(nameNodeRpcAddress)) {
-                                throw new BrokerException(TBrokerOperationStatusCode.INVALID_ARGUMENT,
-                                        "missed " + nameNodeRpcAddress + " configuration");
-                            } else {
-                                conf.set(nameNodeRpcAddress, properties.get(nameNodeRpcAddress));
+                    logger.info("dfs.name.services:" + dfsNameServices);
+                    String[] dfsNameServiceArray = dfsNameServices.split("\\s*,\\s*");
+                    for (String dfsNameService: dfsNameServiceArray) {
```
Comment (wyb) [inline fs_brokers/apache_hdfs_broker/src/main/java/com/starrocks/broker/hdfs/FileSystemManager.java:396]:
```suggestion
                    for (String dfsNameService : dfsNameServiceArray) {
```

**Review by Astralidea — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wyb** [comment]
@Mergifyio backport branch-3.1

**Comment by wyb** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**